src/package-linker.js

```diff
@@ -468,6 +479,16 @@ export default class PackageLinker {
     return range === '*' || satisfiesWithPreleases(version, range, this.config.looseSemver);
   }
 
+  _isChildOfLinkedDep(linkedDepLocations: Array<string>, location: string): boolean {
+    let isWithinLink = false;
+    linkedDepLocations.forEach(loc => {
+      if (location.startsWith(loc)) {
```

Comment:
Also, I will admit that this isn't my favorite way to solve this problem, but it is the least intrusive.

There's no real concept, that I could tell, of "give me the parent PackageReference that asked for this dependency". There is a `requests` array, but in the scenario where package-a (a non-linked dep) _and_ package-b (a link:// dep) requests `foo-bar`, it's not clear to me how, within the context of looping through the `flatTree` data structure, how we would tell which one of those "requests" applies.

Seeing as the flat tree structure provides us with an absolute path (that is not resolved into a symlink), this seemed like a reasonable (albeit slightly hacky) way to determine whether or not it was a child.

Curious to hear thoughts here or if I missed something in the underlying data structures.